src/main/java/com/icthh/xm/ms/dashboard/repository/impl/DefaultWidgetRepositoryWrapper.java

```diff
@@ -48,24 +50,43 @@ public void deleteById(Long id) {
 
     @Override
     public Page<Map<String, Object>> findAuditsById(Long id, Pageable pageable) {
+        Sort.Order order = getPageOrder(pageable);
         AuditQuery auditQuery = auditReader.createQuery()
             .forRevisionsOfEntity(Widget.class,false, true)
             .setFirstResult(pageable.getPageNumber() * pageable.getPageSize())
             .setMaxResults(pageable.getPageSize())
-            .add(AuditEntity.property("id").eq(id));
-        return getResult(auditQuery);
+            .add(AuditEntity.property("id").eq(id))
+            .addOrder(new PropertyAuditOrder(null, new EntityPropertyName(order.getProperty()), order.isAscending()));
+        Long totalCountElements = (Long) auditReader.createQuery().forRevisionsOfEntity(Widget.class, false,true)
+            .addProjection(AuditEntity.revisionNumber().count())
+            .getSingleResult();
+        return getResult(auditQuery, pageable, totalCountElements);
     }
 
     @Override
     public Page<Map<String, Object>> findAllAudits(Pageable pageable) {
+        Sort.Order order = getPageOrder(pageable);
         AuditQuery auditQuery = auditReader.createQuery()
             .forRevisionsOfEntity(Widget.class,false, true)
             .setFirstResult(pageable.getPageNumber() * pageable.getPageSize())
-            .setMaxResults(pageable.getPageSize());
-        return getResult(auditQuery);
+            .setMaxResults(pageable.getPageSize())
+            .addOrder(new PropertyAuditOrder(null, new EntityPropertyName(order.getProperty()), order.isAscending()));
+        Long totalCountElements = (Long) auditReader.createQuery().forRevisionsOfEntity(Widget.class, false,true)
+            .addProjection(AuditEntity.revisionNumber().count())
+            .getSingleResult();
+        return getResult(auditQuery, pageable, totalCountElements);
+    }
+
+    private Sort.Order getPageOrder(Pageable pageable) {
+        Sort.Order order = pageable.getSortOr(Sort.by("originalId.revision.revtstmp").ascending()).stream().findFirst().get();
```

Comment:
pls move to constant with name that allow to understand logic what is going on in this method